yt_dlp/extractor/odkmedia.py

```diff
@@ -0,0 +1,139 @@
+            episodeNumber: $episodeNumber
+            kind: $kind
+            part: $part
+        ) {
+            ...EpisodeDetail
+            __typename
+            }
+        }
+
+        fragment EpisodeDetail on Episode {
+            id
+            episodeNumber
+            part
+            kind
+            link
+            title
+            titleEn
+            titleKo
+            titleZhHans
+            titleZhHant
+            synopsis
+            synopsisEn
+            synopsisKo
+            synopsisZhHans
+            synopsisZhHant
+            seoSynopsis
+            seoKeyword
+            videoCcLanguages
+            videoDuration
+            releaseDate
+            people {
+                ...PeopleItem
+                __typename
+            }
+            images {
+                thumbnail
+                __typename
+            }
+            ageRating {
+                isBlock
+                __typename
+            }
+            __typename
+            }
+            fragment PeopleItem on People {
+                role
+                slug
+                nameEn
+                nameZhHans
+                nameZhHant
+                __typename
+            }
+            """
+
+    def _real_extract(self, url):
+        program_slug, display_id, ep_number = self._match_valid_url(url).group(
+            'series_name', 'id', 'episode_num')
+        webpage = self._download_webpage(url, display_id)
+
+        video_info = self._download_json(
+            'https://odc-graphql.odkmedia.io/graphql', display_id,
+            headers={'Content-type': 'application/json'},
+            data=json.dumps({
+                'operationName': 'Episode',
+                'variables': {
+                    'programSlug': program_slug,
+                    'episodeNumber': int(ep_number),
+                    'kind': 'series',
+                    'part': None
+                },
+                'query': self._QUERY}
+            ).encode())['data']['episode']
+
+        try:
+            source_json = self._download_json(
+                f'https://odkmedia.io/odc/api/v2/playback/{video_info["id"]}/', display_id,
+                headers={'Authorization': '', 'service-name': 'odc'})
+        except ExtractorError as e:
+            if (isinstance(e.cause, urllib.error.HTTPError)):
+                error_data = self._parse_json(e.cause.read(), display_id)['detail']
+                raise GeoRestrictedError(error_data)
+
+        formats, subtitles = [], {}
+        for source in traverse_obj(source_json, ('sources', ...)):
+            if source.get('type') == 'hls':
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(source.get('url'), display_id)
+                formats.extend(fmts)
+                self._merge_subtitles(subs, target=subtitles)
+            else:
+                self.report_warning('Can\'t extract format other than hls', display_id)
```

Comment:
we could put the `type` value in the warning so that users' logs are more useful to us:
```suggestion
                self.report_warning(f'Skipping extraction of format: {source.get("type")}', display_id)
```